tests/benchmarks/iam/functions.tql

```diff
@@ -0,0 +1,121 @@
+define
+# utils
+
+
+fun noop_bind_subject($s: subject) -> {subject}:
+match $s isa subject;
+return { $s };
+
+fun noop_bind_object($o: object) -> {object}:
+match $o isa object;
+return { $o };
+
+fun noop_bind_action($a: action) -> {action}:
+match $a isa action;
+return { $a };
+
+
+# memberships
+fun has_group_membership($member: subject) -> { user-group }:
+match
+    let $group in noop_bind_subject($group-inner);
+    $group isa user-group;
+    {
+        $m isa group-membership, links (group: $group-inner, member: $member);
+    } or {
+        $m1 isa group-membership, links (group: $group-inner, member: $intermediate);
+        let $intermediate in has_group_membership($member);
+    };
+return { $group };
+
+
+fun has_collection_membership($member: object) -> { resource-collection }:
+match
+    let $collection in noop_bind_object($collection-inner);
```

Comment:
how come you didn't just do `$collection isa $object;`?

Reply:
It's a trick I used to use to force it being done after the disjunction.

Reply:
Let me know if removing them works, I'm not a fan at all

Reply:
Ok, maybe you can resolve this with Dmitrii before we merge hopefully tomorrow!